### ISSUE TITLE
Amplify not recognizing my artifact directory during deployment.

### ISSUE THREAD

**Issue by slhernandez** [aws-amplify/amplify-console#313]
appId: d1ms7jt11oekyc
region: us-east-1

Hello,

I think I have my build settings incorrect.  I am having trouble getting my build (baseDirectory) recognized by the deploy step.  I am deploying a vanilla react.js app with a typical "npm run build" script that creates a "build" folder.  For some reason, Amplify can't locate the build folder even though it's been generated.  I even inserted some postBuild commands to ls la the directory contents and confirm the build directory is there.  

Below is my log file.  

[amplify_output_slh.txt](https://github.com/aws-amplify/amplify-console/files/3962741/amplify_output_slh.txt)

Any insight to this is greatly appreciated.

**Comment by kahdojay** [comment]
hi @slhernandez can you share your buildspec?

**Comment by slhernandez** [comment]
Hi @kahdojay.  I am no longer using Amplify.  Decided to go in a different direction with my deployment.  I am going to close this issue.  Thanks!

**Comment by swaminator** [comment]
@slhernandez sorry to see you go. Any particular reason why you moved away?

**Comment by gvnttdata** [comment]
I am having this exact issue, I am trying to host a Nuxt project and for some reason I see the build script being run but not being able to be found right after.

this is what I'm getting in my logs:

```
# Starting phase: build
# Executing command: npm run build
xxxxx@1.0.0 build /codebuild/output/xxx/src/xxxx
nuxt build
Completed phase: build
## Build completed successfully
...
!!! Error: Artifact directory doesn't exist: dist
```

and this is my build instructions
```
version: 0.1
frontend:
  phases:
    preBuild:
      commands:
        - npm ci
    build:
      commands:
        - npm run build
  artifacts:
    baseDirectory: dist
    files:
      - '**/*'
  cache:
    paths: []
```

Any idea where this is going wrong? This is just a basic nuxt project, I did not think starting with Amplify would be such an investment.

**Comment by gvnttdata** [comment]
I tried changing my baseDirectory to '.nuxt' and the build succeeded but I'm getting the error "xxxxx redirected you too many times." This is still the initial setup so I have yet to see my page actually work so I'm not sure if this made things better or worse but it might be a step in the right direction.

**Comment by koprinkov** [comment]
My problem was that variables wasn't recognised in artifacts section and I had a 'cd' command in preBuild. So adding full path it worked.

version: 1.0
env:
  variables:
      BUILD_FOLDER: app
      OUTPUT_FOLDER: dist
frontend:
    phases:
        preBuild:
            commands:
                - cd ${BUILD_FOLDER}
                - mkdir ./${OUTPUT_FOLDER}
                - npm install
                - npm audit fix
        build:
            commands:
                - npm run build-release
        postBuild:
            commands:
                - pwd
                - ls ${OUTPUT_FOLDER}
    artifacts:
        baseDirectory: /app/dist
        files:
            - '**/*'
    cache:
        paths:
            - node_modules/**/*

**Comment by thisbenrogers** [comment]
@koprinkov thank you, this worked for my team.

**Comment by tcoz** [comment]
I ran headfirst into this today. I updated Angular 11 to Angular 12 (locally), made no other changes, tested my serve and build commands, everything worked the same. There were no code changes, etc. I didn't expect any problems as the same exact configuration had worked many times before (the website is and has been live). 

I committed the package.json changes, and the build failed, artifact directory not found. I tried everything basic I could think of (mostly altering the paths to things that might make sense), even committing the build directory in the place it was supposed to be. 

There are no cd commands, it's very basic build. But I now have a web app that won't deploy because no matter what I try, an artifact directory that used to be fine can no longer be found.
